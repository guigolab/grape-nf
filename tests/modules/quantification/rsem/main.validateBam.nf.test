nextflow_process {

    name "Test Process validate"
    script "modules/quantification/rsem/main.nf"
    process "validateBam"

    tag "module"

    test("Should validate BAM file for RSEM") {

        when {
            process {
                """
                input[0] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toTranscriptome.bam"), "bam", "TranscriptomeAlignments", true, "MATE2_SENSE" ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should not validate BAM file for RSEM") {

        when {
            process {
                """
                input[0] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true, "MATE2_SENSE" ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
