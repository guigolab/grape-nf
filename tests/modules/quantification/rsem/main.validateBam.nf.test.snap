{
    "Should not validate BAM file for RSEM": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        ".\nSkipped region is detected (cigar N) for read HWI-ST985:81:C0C5PACXX:8:1104:20582:12084!\nTo use RSEM, please align your reads to a set of transcript sequences instead of a genome.\nThe input file is not valid!\n"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-03T12:30:00.196523"
    },
    "Should validate BAM file for RSEM": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        ".\nThe input file is valid!\n"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-03T12:29:51.499688"
    }
}